Source/CPack/cmCPackDebGenerator.cxx

```diff
@@ -411,7 +417,28 @@ int cmCPackDebGenerator::createDeb()
     out << std::endl;
     }
 
-  const std::string strGenWDIR(this->GetOption("GEN_WDIR"));
+  const std::string shlibsfilename = strGenWDIR + "/shlibs";
+
+  const char* debian_pkg_shlibs = this->GetOption("CPACK_DEBIAN_PACKAGE_SHLIBS");
+  if(this->IsOn("CPACK_DEBIAN_PACKAGE_GENERATE_SHLIBS") && debian_pkg_shlibs && strlen(debian_pkg_shlibs))
```

Comment:
This is a very inefficient way of checking if the string is non-empty. How about `*debian_pkg_shlibs`?